src/Bundle/Migration/CodeGenerator.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\GridBundle\Migration;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Param;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\Node\Stmt\UseUse;
+
+final class CodeGenerator
+{
+    /** @var array<string> */
+    private array $classesToUse = [];
+
+    /** @var array<Node> */
+    private array $classNodes = [];
+
+    private ?Node $namespace = null;
+
+    public function __construct(
+        private CodeOutputter $codeOutputter,
+    ) {
+    }
+
+    public function setNamespace(string $namespace): void
+    {
+        $this->namespace = new Node\Stmt\Namespace_(new Name($namespace));
+    }
+
+    public function addUseStatement(string $classToUse): void
+    {
+        $this->classesToUse[] = $classToUse;
+    }
+
+    /**
+     * If you want to use a reference to a class but don't want to use the FQN everywhere you
+     * need an import.
+     * This function adds an import for the class and returns the imported name back
+     *
+     * @param class-string $className
+     */
+    public function getRelativeClassName(string $className): Name
+    {
+        $this->addUseStatement($className);
+
+        $namespaceParts = explode('\\', $className);
+
+        return new Name(end($namespaceParts));
+    }
+
+    public function addClass(string $className, ?string $extends, array $implements = [], array $body = []): void
+    {
+        $classConfiguration = ['stmts' => $body];
+
+        if ($extends !== null) {
```

Comment:
Maybe use Yoda style

Reply:
For this it would probably be the best to add a check in the codestyle of the plugin for that. But I can fix those as well.